astropy/modeling/bounding_box.py

```diff
@@ -578,47 +632,116 @@ class ModelBoundingBox(_BoundingDomain):
         (default), 'F': Fortran/mathematical notation order, e.g. x, y, z.
     """
 
-    def __init__(self, intervals: Dict[int, _Interval], model,
-                 ignored: List[int] = None, order: str = 'C'):
+    def __init__(self, intervals: Dict[str, _Interval], model = None,
+                 ignored: List[str] = None, order: str = 'C'):
+
+        # HACK, prevents a huge number of errors in legacy bounding box related tests.
+        #       should be comparing to () but there is a bug in __eq__.
+        if isinstance(intervals, tuple) and len(intervals) == 0:
+            intervals = {}
+
+        self._intervals = intervals
         super().__init__(model, ignored, order)
 
+    def _pop_intervals(self):
+        intervals = self._intervals
         self._intervals = {}
-        if intervals != () and intervals != {}:
-            self._validate(intervals, order=order)
 
-    def copy(self, ignored=None):
-        intervals = {index: interval.copy()
-                     for index, interval in self._intervals.items()}
+        return intervals
 
-        if ignored is None:
-            ignored = self._ignored.copy()
+    def _verify_intervals_dict(self, _external_ignored: List[str] = None):
+        if _external_ignored is None:
+            _external_ignored = []
+
+        intervals = self._pop_intervals()
+
+        for key, value in intervals.items():
+            name = self._get_name(key)
+            if name in _external_ignored:
+                raise ValueError(f"Interval: {name} is being externally ignored!")
```

Comment:
Will it cause trouble that we leave self._intervals empty after hitting this error?  Ordinarily I'd expect that a validation failure would reject the change and leave the object as it was.

Reply:
I believe this is addressed by 0dfa0c3